frontend/src/components/editor/chrome/wrapper/footer-items/runtime-settings.tsx

```diff
@@ -0,0 +1,346 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import {
+  ChevronDownIcon,
+  PowerOffIcon,
+  ZapIcon,
+  ZapOffIcon,
+} from "lucide-react";
+import type React from "react";
+import { useEffect, useState } from "react";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { Switch } from "@/components/ui/switch";
+import { useResolvedMarimoConfig, useUserConfig } from "@/core/config/config";
+import type { UserConfig } from "@/core/config/config-schema";
+import { saveUserConfig } from "@/core/network/requests";
+import { isWasm } from "@/core/wasm/utils";
+import { cn } from "@/utils/cn";
+import { FooterItem } from "../footer-item";
+
+interface RuntimeSettingsProps {
+  className?: string;
+}
+
+export const RuntimeSettings: React.FC<RuntimeSettingsProps> = ({
+  className,
+}) => {
+  const [userConfig, setUserConfig] = useUserConfig();
+  const config = useResolvedMarimoConfig()[0];
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+
+  // Close dropdown when screen size crosses the md breakpoint
+  useEffect(() => {
+    // 768px matches Tailwind's 'md' breakpoint used in hidden md:flex
+    // TODO: When upgrading to Tailwind v4, use var(--breakpoint-md)
+    const mediaQuery = window.matchMedia("(min-width: 768px)");
+    const handleChange = () => {
+      if (mediaQuery.matches) {
+        setDropdownOpen(false);
+      }
+    };
+
+    mediaQuery.addEventListener("change", handleChange);
+    return () => mediaQuery.removeEventListener("change", handleChange);
+  }, []);
+
+  const handleStartupToggle = async (checked: boolean) => {
+    const newConfig = {
+      ...config,
+      runtime: {
+        ...config.runtime,
+        auto_instantiate: checked,
+      },
+    };
+    await saveUserConfig({ config: newConfig }).then(() =>
+      setUserConfig(newConfig),
+    );
+  };
+
+  const handleCellChangeToggle = async (checked: boolean) => {
+    const newUserConfig: UserConfig = {
+      ...userConfig,
+      runtime: {
+        ...userConfig.runtime,
+        on_cell_change: checked ? "autorun" : "lazy",
+      },
+    };
+    await saveUserConfig({ config: newUserConfig }).then(() =>
+      setUserConfig(newUserConfig),
+    );
+  };
+
+  const handleModuleReloadChange = async (
+    option: "off" | "lazy" | "autorun",
+  ) => {
+    const newUserConfig: UserConfig = {
+      ...userConfig,
+      runtime: {
+        ...userConfig.runtime,
+        auto_reload: option,
+      },
+    };
+    await saveUserConfig({ config: newUserConfig }).then(() =>
+      setUserConfig(newUserConfig),
+    );
+  };
+
+  // Check if all reactivity is disabled
+  const allReactivityDisabled =
+    !config.runtime.auto_instantiate &&
+    config.runtime.on_cell_change === "lazy" &&
+    (isWasm() || config.runtime.auto_reload !== "autorun");
```

Comment:
The icon will show with strikethrough if all are disabled... and active if any of the settings are enabled.